src/net_processing.cpp

```diff
@@ -3852,8 +3861,14 @@ bool PeerManager::ProcessMessages(CNode* pfrom, std::atomic<bool>& interruptMsgP
 
     // this maintains the order of responses
     // and prevents vRecvGetData to grow unbounded
-    if (!pfrom->vRecvGetData.empty()) return true;
-    if (!pfrom->orphan_work_set.empty()) return true;
+    {
+        LOCK(pfrom->cs_vRecv);
+        if (!pfrom->vRecvGetData.empty()) return true;
+    }
+    {
+        LOCK2(cs_main, g_cs_orphans);
```

Comment:
I don't think `cs_main` is required here.

Reply:
right!